librz/core/cmd/cmd_cmp.c

```diff
@@ -4,17 +4,20 @@
 #include <rz_cmp.h>
 #include "../core_private.h"
 
-static int rizin_compare_words(RzCore *core, ut64 of, ut64 od, int len, int ws) {
+static void rizin_compare_words(RzCore *core, ut64 of, ut64 od, int len, int ws) {
+	rz_return_if_fail(core && (ws == 1 || ws == 2 || ws == 4 || ws == 8));
 	int i;
 	bool useColor = rz_config_get_i(core->config, "scr.color") != 0;
-	utAny v0, v1;
+	bool big_endian = rz_config_get_b(core->config, "cfg.bigendian");
+	ut64 v[2];
 	RzConsPrintablePalette *pal = &rz_cons_singleton()->context->pal;
 	for (i = 0; i < len; i += ws) {
-		memset(&v0, 0, sizeof(v0));
-		memset(&v1, 0, sizeof(v1));
-		rz_io_nread_at(core->io, of + i, (ut8 *)&v0, ws);
-		rz_io_nread_at(core->io, od + i, (ut8 *)&v1, ws);
-		char ch = (v0.v64 == v1.v64) ? '=' : '!';
+		for (size_t j = 0; j < 2; j++) {
+			ut8 tmp[8] = { 0 };
+			rz_io_nread_at(core->io, (j ? od : of) + i, tmp, ws);
+			v[j] = rz_read_ble(tmp, big_endian, ws * 8);
+		}
```

Comment:
isn't easier to just read 16 bytes directly once and call rz_read_ble twice?

Reply:
`od` and `of` can be anywhere, they are not contiguous.